### ISSUE TITLE
Display names and profile pictures of some users are not fetched in group chats

### ISSUE THREAD

**Issue by pavloburykh** [status-im/status-react#17230]
**Preconditions:** 

1. User A is mutual contact with User B and User C. 
2. User B and User C are not mutual contacts

**Steps:** 

1. User A creates group chat and invites User B and User C
2. User B  **goes offline**
3. User C opens group chat
4. Open mentions list by User C
5. See which User's B data is fetched for User C

**Actual result:**

1. Display name and profile picture of User B are not fetched for User C

User's B data is fetched for User C in one of the following cases:

1. User C opens profile page of User B
2. User B goes back online
3. There is a message from User B in chat which is fetched by User C

On the video below you can observe the described behaviour (User C is from the right part of the screen)

[User_C_logs.zip](https://github.com/status-im/status-mobile/files/12559725/User_C_logs.zip)

https://github.com/status-im/status-mobile/assets/97245802/392fedc6-3cac-4ad2-8fdd-dfa43fab6e62

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

**Comment by cammellos** [comment]
This is a status-go change most likely

**Comment by mohsen-ghafouri** [comment]
Hey @pavloburykh i did multiple test but couldn't reproduce the issue in latest dev branch. here is what i did, maybe you can give me more hint to reproduce the issue correctly 

https://github.com/status-im/status-mobile/assets/71308738/4702b953-f443-4b93-b4b2-2a88bbb1e9c4

**Comment by pavloburykh** [comment]
> Hey @pavloburykh i did multiple test but couldn't reproduce the issue in latest dev branch. here is what i did, maybe you can give me more hint to reproduce the issue correctly

Hey @mohsen-ghafouri! Thank you for looking at this bug. I have just found out that I have added wrong video to this issue description. My video is demonstrating  similar problem we are facing in communities which is actually logged separately here https://github.com/status-im/status-mobile/issues/17229 I am sorry for this confusion. 

So, the current issue exclusively relates only to group chats. As group chats are out of priority now I think we can leave this issue for the future. 

In case with group chats I am still reproducing the issue following exact steps from this ticket


https://github.com/status-im/status-mobile/assets/97245802/90a6d022-f7fb-42e7-98d9-034fd70b3a29



Speaking about communities: I have failed to reproduce the issue following previously described steps. But you still can easily reproduce the issue if you restore your main Status account (which is member of Status community) on mobile, go to any channel in Status community and check mentions list. You will see bunch of users with pub keys with absent profile icons etc. I have described it in this issue https://github.com/status-im/status-mobile/issues/17229
